client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -152,38 +185,46 @@ class CalypsoifyIframe extends Component {
 			}
 		}
 
-		if ( 'trashPost' === action ) {
+		if ( EditorActions.TrashPost === action ) {
 			const { siteId, editedPostId, postTypeTrashUrl } = this.props;
 			this.props.trashPost( siteId, editedPostId );
 			this.props.navigate( postTypeTrashUrl );
 		}
 
-		if ( 'goToAllPosts' === action ) {
+		if ( EditorActions.GoToAllPosts === action ) {
 			this.props.navigate( this.props.allPostsUrl );
 		}
 
-		if ( 'openRevisions' === action ) {
+		if ( EditorActions.OpenRevisions === action ) {
 			this.props.openPostRevisionsDialog();
 		}
 
-		if ( 'previewPost' === action ) {
+		if ( EditorActions.PreviewPost === action ) {
 			const { postUrl } = payload;
 			this.openPreviewModal( { postUrl, previewPort: ports[ 0 ] } );
 		}
 	};
 
-	loadRevision = revision => {
+	loadRevision = ( {
+		post_title: title,
+		post_excerpt: excerpt,
+		post_content: content,
+	}: {
+		post_title: string;
+		post_excerpt: string;
+		post_content: string;
+	} ) => {
```

Comment:
Creating a `PostRevision` type would let us avoid this unwieldy destructuring.

Reply:
I thought about it but felt like creating a `PostRevision` would both be too restrictive and something I don't know if I need. Were we to have a full `PostRevision` model, or if I knew it, I would have reused it here. that being said, we only need these three props so I left the restructuring in.